it/controllers/registration/applicant/SoleTraderIdentificationControllerISpec.scala

```diff
@@ -97,40 +105,48 @@ class SoleTraderIdentificationControllerISpec extends ControllerISpec {
 
       whenReady(res) { result =>
         result.status mustBe SEE_OTHER
-        result.headers(LOCATION) must contain(applicantRoutes.CaptureRoleInTheBusinessController.show().url)
+        result.headers(LOCATION) must contain(applicantRoutes.FormerNameController.show().url)
       }
     }
 
-    "redirect to the CaptureRoleInTheBusiness page for a sole trader journey" in new Setup {
+    "redirect to the RoleInTheBusiness page if the user is not a Sole Trader" in new Setup {
       given()
         .user.isAuthorised
         .audit.writesAudit()
         .audit.writesAuditMerged()
-        .s4lContainer[ApplicantDetails].contains(validFullApplicantDetails)
-        .s4lContainer[ApplicantDetails].clearedByKey
-        .vatScheme.isUpdatedWith(validFullApplicantDetails.copy(
-        entity = Some(testSoleTrader)
-      ))
+        .vatScheme.has("applicant-details", Json.toJson(ApplicantDetails()))
+        .s4lContainer[ApplicantDetails].isUpdatedWith(ApplicantDetails())
+        .vatScheme.contains(
+        VatScheme(id = currentProfile.registrationId,
+          status = VatRegStatus.draft,
+          eligibilitySubmissionData = Some(testEligibilitySubmissionData.copy(partyType = UkCompany))
+        )
+      )
 
       stubGet(retrieveDetailsUrl, OK, testSTIResponse.toString)
       insertCurrentProfileIntoDb(currentProfile, sessionId)
 
-      val res: Future[WSResponse] = buildClient(s"/register-for-vat/sti-callback?journeyId=$testJourneyId").get()
+      val res = buildClient(s"/register-for-vat/sti-callback?journeyId=$testJourneyId").get()
 
       whenReady(res) { result =>
         result.status mustBe SEE_OTHER
         result.headers(LOCATION) must contain(applicantRoutes.CaptureRoleInTheBusinessController.show().url)
       }
     }
 
-    "redirect to the CaptureRoleInTheBusiness page when the model in S4l is full" in new Setup {
+    "redirect to the RoleInTheBusiness page when the model in S4l is full and the user is not a Sole Trader" in new Setup {
       given()
         .user.isAuthorised
         .audit.writesAudit()
         .audit.writesAuditMerged()
-        .s4lContainer[ApplicantDetails].contains(validFullApplicantDetails)
-        .s4lContainer[ApplicantDetails].clearedByKey
-        .vatScheme.isUpdatedWith(validFullApplicantDetails)
+        .vatScheme.has("applicant-details", Json.toJson(ApplicantDetails()))
```

Comment:
This and the test below are meant to be testing that the service submits to backend when S4L contains valid applicant details, so this should stay
        .s4lContainer[ApplicantDetails].contains(validFullApplicantDetails)
        .s4lContainer[ApplicantDetails].clearedByKey
        .vatScheme.isUpdatedWith(validFullApplicantDetails)